### PR TITLE
Adi cors fix

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

**Review by jayras-o — APPROVED** [review]

File: src/test/scala/com/webtrends/harness/component/akkahttp/RouteGeneratorTest.scala [hunk 1]
```diff
@@ -226,7 +227,16 @@ class RouteGeneratorTest extends WordSpec with ScalatestRouteTest with Predefine
 
   "Cors settings" should {
     val whiteListOrigin = HttpOrigin("http://example.com")
-    val corsSettings = CorsSettings.defaultSettings
+    val corsSettings = CorsSettings(ConfigFactory.parseString(
```
Comment (Kraagen) [inline src/test/scala/com/webtrends/harness/component/akkahttp/RouteGeneratorTest.scala]: There is no need to use json or typesafe config here.

You can just use the CorsSettings constructor or helper "with" methods to construct it.

File: src/main/scala/com/webtrends/harness/component/akkahttp/routes/RouteGenerator.scala [hunk 2]
```diff
@@ -233,7 +233,7 @@ object RouteGenerator {
                           accessLogIdGetter: Option[AkkaHttpRequest => String]): Directive0 =
     corsSettings match {
       case Some(cors) => handleRejections(corsRejectionHandler(request, accessLogIdGetter)) &
-        CorsDirectives.cors(cors.withAllowedMethods(immutable.Seq(method)))
+        CorsDirectives.cors(cors.withAllowedMethods(cors.allowedMethods ++ immutable.Seq(method)))
```
Comment (Kraagen) [inline src/main/scala/com/webtrends/harness/component/akkahttp/routes/RouteGenerator.scala]: This list should be distinct.

File: src/test/scala/com/webtrends/harness/component/akkahttp/RouteGeneratorTest.scala (hunk 1 above)
Comment (ghost) [reply src/test/scala/com/webtrends/harness/component/akkahttp/RouteGeneratorTest.scala]: Updated with empty list

**Review by Kraagen — APPROVED** [review]

File: src/main/scala/com/webtrends/harness/component/akkahttp/routes/RouteGenerator.scala (hunk 2 above)
Comment (ghost) [reply src/main/scala/com/webtrends/harness/component/akkahttp/routes/RouteGenerator.scala]: Added distinct.

**Pull request merged**